packages/contracts/contracts/delegateTargets/VirtualAppAgreement.sol

```diff
@@ -12,47 +12,49 @@ contract VirtualAppAgreement {
   using Transfer for Transfer.Terms;
 
   struct Agreement {
-    Registry registry;
-    NonceRegistry nonceRegistry;
+    address registryAddr;
```

Comment:
I think we might be able to store this as a constant value on the contract since it will be known at migration time

Reply:
how is this done?

Reply:
I'm guessing at the bytecode level there would be `__registryAddr__` at 2-3 places in the code that gets replaced at link time

Reply:
I'm not 100% sure in the case of delegatecalls because I don't know how `constant`s work in Solidity precisely but it seems like you should be able to do this

Reply:
well just doing `  address constant REGISTRY_ADDR = __registryAddr__` failed with a syntax error; will mark as TODO since I have no idea how to do this